Solutions/Marain.Operations.OpenApi/Marain/Operations/OpenApi/OperationsControl.yaml

```diff
@@ -250,7 +250,7 @@ components:
     Operation:
       description: A long-running operation
       type: object
-      required: ["createdDateTime", "id", "lastActionDateTime", "contentId", "percentComplete", "resourceLocation", "status", "tenantId"]
+      required: ["createdDateTime", "id", "lastActionDateTime", "status", "tenantId"]
```

Comment:
It looks like it was "very close to a bug"; we we previously requiring them, but setting them to `string.Empty` rather than null. And this is the fix (explicit nullability).

Reply:
The changes in this YAML were partly motivated by the fact that the two definitions of the Operation type in the two different YAML files had drifted apart, even though I believe they are meant to represent the same type. (They are certainly represented by the same type inside the service implementations.)

But yes, there was also an element of the "Empty string as `null` in disguise" antipattern.

There as also some inconsistency. The code previously used a mixture of `null` and `""` to represent "not set". The schemas sometimes made optional properties explicitly optional, but sometimes made them `required` but nullable.

So I've taken this opportunity to make things more consistent:

* in the code, we now use nullability to denote the idea that a value might not be present
* in the YAML, optional properties are not listed in `required`, but we tolerate them being present and explicitly set to `null` in the spirit of Postel's principle of robustness